core/io/logger.cpp

```diff
@@ -180,6 +182,21 @@ RotatedFileLogger::RotatedFileLogger(const String &p_base_path, int p_max_files)
 		base_path(p_base_path.simplify_path()),
 		max_files(p_max_files > 0 ? p_max_files : 1) {
 	rotate_file();
+
+#ifdef MODULE_REGEX_ENABLED
+	strip_ansi_regex = memnew(RegEx);
+	strip_ansi_regex->compile("\u001b\\[((?:\\d|;)*)([a-zA-Z])");
+#endif // MODULE_REGEX_ENABLED
+}
+
+RotatedFileLogger::~RotatedFileLogger() {
+	// FIXME: This doesn't appear to be run when exiting the project by clicking the X window button,
+	// therefore a leaked instance issue occurs:
+	//
+	//     WARNING: ObjectDB instances leaked at exit (run with --verbose for details).
+	//     at: cleanup (./core/object/object.cpp:2290)
+	//     Leaked instance: RegEx:9223372039421689985
+	memdelete(strip_ansi_regex);
```

Comment:
I'm running into an issue reported by asan here, not sure how to fix it. If I put a print statement in the destructor, I never see it. Logger and CompositeLogger both have destructors which seem to work.

I just tried making the destructor virtual (like the one for StdLogger) to no avail.

Reply:
Loggers are deleted in the `OS` destructor, which is called after `ObjectDB` deletion. `RotatedFileLogger` is already using `detach_from_objectdb()` method on `file` right after creation to remove it for the `ObjectDB` and avoid errors on exit, `RegEx` probably should do the same.

Reply:
Also, `RegEx` is `RefCounted`, so it can be declared `Ref<RegEx>` to avoid manual deletion and need for destructor.

Reply:
It works, thanks :slightly_smiling_face: